implementation/server-undertow/src/main/java/net/opentsdb/tsd/TSDMain.java

```diff
@@ -229,15 +243,24 @@ public static void main(final String[] args) {
     manager.deploy();
     
     try {
-      final PathHandler path = Handlers.path(Handlers.redirect(root))
+      HttpHandler handler = Handlers.path(Handlers.redirect(root))
               .addPrefixPath(root, manager.start());
       
-      final HttpHandler encoding_handler= new EncodingHandler.Builder()
+      if (!Strings.isNullOrEmpty(tsdb.getConfig().getString(CORS_PATTERN_KEY))) {
+        handler = new com.stijndewitt.undertow.cors.Filter(handler);
+        ((com.stijndewitt.undertow.cors.Filter) handler).setUrlPattern(
+            tsdb.getConfig().getString(CORS_PATTERN_KEY));
+        ((com.stijndewitt.undertow.cors.Filter) handler).setAllowHeaders(
+            tsdb.getConfig().getString(CORS_HEADERS_KEY));
+      }
+      
+      // support compression!
+      handler = new EncodingHandler.Builder()
```

Comment:
Might be a good idea to expose the Undertow IO/workertask threads as a config

Reply:
yeah I'll get a ton of configs in a future pr.

Reply:
Right on!